app/assets/javascripts/modal/modal-workflow.js

```diff
@@ -26,6 +26,7 @@ ModalWorkflow.prototype.render = function (response) {
 
 ModalWorkflow.prototype.renderSuccess = function (result) {
   this.$multiSectionViewer.showDynamicSection(result.body)
+  this.setModalLabel()
```

Comment:
I'd be tempted to rename this to `setModalAriaLabel` so it's clear we're not dealing with `<label>` elements.